src/cache/cache_las.c

```diff
@@ -838,12 +875,23 @@ __wt_las_insert_block(WT_CURSOR *cursor,
 
 	__las_restore_isolation(session, saved_isolation);
 
+	if (ret == 0 && insert_cnt > 0)
+		ret = __wt_calloc(session, insert_cnt,
+		    sizeof(WT_BIRTHMARK_DETAILS), &multi->page_las.birthmarks);
+
 	if (ret == 0 && insert_cnt > 0) {
 		multi->page_las.las_pageid = las_pageid;
 		multi->page_las.has_prepares = prepared_insert_cnt > 0;
+		memcpy(multi->page_las.birthmarks, birthmarks->mem,
```

Comment:
Is there any reason we can't just do:

```
multi->page_las.birthmarks = birthmarks->mem;
```

Seems weird that we allocate a new buffer, `memcpy` into it and free the old one immediately.

Reply:
That is a good question, I had tried to only allocate only one buffer before till I realized that scratch buffer is tied up with a session and here the allocated memory is tied up with the page structure that gets accessed through multiple sessions. There is certainly scope of improvement here, and we will likely work on it as a separate ticket.

Reply:
This is right -- we use a scratch buffer and grow it fast if necessary (`__wt_buf_extend()` allocates chunks of memory, not just what's a requirement at the moment), and this copies into just what we need. To just save off the scratch buffer's memory might waste a huge chunk of space, if the scratch buffer is big.